pkg/analyze/velero.go

```diff
@@ -0,0 +1,741 @@
+package analyzer
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	appsV1 "k8s.io/api/apps/v1"
+
+	restic_types "github.com/replicatedhq/troubleshoot/pkg/analyze/types"
+	"golang.org/x/mod/semver"
+
+	"github.com/pkg/errors"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+)
+
+type AnalyzeVelero struct {
+	analyzer *troubleshootv1beta2.VeleroAnalyze
+}
+
+func (a *AnalyzeVelero) Title() string {
+	title := a.analyzer.CheckName
+	if title == "" {
+		title = "Velero"
+	}
+
+	return title
+}
+
+func (a *AnalyzeVelero) IsExcluded() (bool, error) {
+	return isExcluded(a.analyzer.Exclude)
+}
+
+func (a *AnalyzeVelero) Analyze(getFile getCollectedFileContents, findFiles getChildCollectedFileContents) ([]*AnalyzeResult, error) {
+	results, err := a.veleroStatus(a.analyzer, getFile, findFiles)
+	if err != nil {
+		return nil, err
+	}
+	for i := range results {
+		results[i].Strict = a.analyzer.Strict.BoolOrDefaultFalse()
+	}
+	return results, nil
+}
+
+func (a *AnalyzeVelero) veleroStatus(analyzer *troubleshootv1beta2.VeleroAnalyze, getFileContents getCollectedFileContents, findFiles getChildCollectedFileContents) ([]*AnalyzeResult, error) {
+	excludeFiles := []string{}
+	results := []*AnalyzeResult{}
+
+	oldVeleroRepoType := false
+	veleroVersion, err := getVeleroVersion(excludeFiles, findFiles)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to find velero deployment")
+	}
+
+	// Check if the version string is valid erer
+	if !semver.IsValid(veleroVersion) {
+		return nil, errors.Errorf("Invalid velero semver: %s", veleroVersion)
+	}
+
+	// check if veleroVersion is less than 1.10.x
+	compareResult := semver.Compare(veleroVersion, "1.10.0")
+	if compareResult < 0 {
+		fmt.Printf("Version %s is less than %s\n", veleroVersion, "1.10.0")
```

Comment:
Log using `klog` as other places in the code. Perhaps `klog.V(3)` or even level 2?

Reply:
didn't need to print this. leftover code.. ty for the find!